astropy/cosmology/tests/test_connect.py

```diff
@@ -151,30 +93,64 @@ def test_from_subclass_complete_info(self, tmpdir, instance, format):
         assert got == cosmo
         assert got.meta == cosmo.meta
 
+    @pytest.mark.skip("TODO: generalize over all save formats for this test.")
+    def test_readwrite_from_subclass_partial_info(self, cosmo, tmpdir):
+        """
+        Test writing from an instance and reading from that class.
+        This requires partial information.
+
+        .. todo::
+
+            - generalize over all save formats for this test.
+            - remove the method defined in subclass
+        """
+
+
+class TestReadWriteCosmology(ReadWriteTestMixin):
+    """Run tests here on CosmologyRead/Write."""
+
+    @pytest.fixture(params=cosmo_instances)
+    def cosmo(self, request):
+        return getattr(cosmology.realizations, request.param)
+
+    # ==============================================================
+
+    @pytest.mark.parametrize("format", readwrite_formats)
+    def test_write_methods_have_explicit_kwarg_overwrite(self, format):
+        writer = io_registry.get_writer(format, Cosmology)
+        # test in signature
+        sig = inspect.signature(writer)
+        assert "overwrite" in sig.parameters
+
+        # also in docstring
+        assert "overwrite : bool" in writer.__doc__
+
+    # @pytest.mark.parametrize("format", readwrite_formats)
     @pytest.mark.parametrize("instance", cosmo_instances)
-    def test_from_subclass_partial_info(self, tmpdir, instance):
+    def test_readwrite_from_subclass_partial_info(self, instance, tmpdir):
         """
         Test writing from an instance and reading from that class.
         This requires partial information.
 
         .. todo::
 
-            generalize over all save formats for this test.
+            remove when fix method in super
```

Comment:
The comment is addressed to yourself, so perhaps not relevant, but I would have no idea what it meant!

Reply:
Ah yeah. I haven't figured out how to do this in general. Modifying a dict is different than a table is different than a Model. I want to test reading from a class and having that class fill in missing information, but I need to figure out how to delete that information. So currently this test only runs on dicts.

Reply:
It's not dire, which is why I'm leaving in a TODO, but it would be best to show the information inference works irrespective of the format.

Reply:
I think I have a solution... you'll see it soonish 😆